gordo_components/cli/cli.py

```diff
@@ -204,15 +234,53 @@ def build(
     model_config = pipeline_into_definition(pipeline_from_definition(model_config))
     logger.debug(f"Fully expanded model config: {model_config}")
 
-    model_location = provide_saved_model(
-        name, model_config, data_config, metadata, output_dir, model_register_dir
-    )
+    if cv_mode == "cross_val_only":
+
+        cache_model_location = None
+        if model_register_dir is not None:
+            cache_key = calculate_model_key(
+                name, model_config, data_config, metadata=metadata
+            )
+            cache_model_location = check_cache(model_register_dir, cache_key)
+
+        if cache_model_location:
+            metadata = load_metadata(cache_model_location)
+        else:
+            _, metadata = build_model(
+                name, model_config, data_config, metadata, cv_mode
+            )
+
+    elif cv_mode == "build_only":
+
+        model_location = provide_saved_model(
+            name,
+            model_config,
+            data_config,
+            metadata,
+            output_dir,
+            model_register_dir,
+            cv_mode=cv_mode,
+        )
+        metadata = load_metadata(model_location)
+
+    else:
+        model_location = provide_saved_model(
+            name,
+            model_config,
+            data_config,
+            metadata,
+            output_dir,
+            model_register_dir,
+            cv_mode=cv_mode,
+        )
+        metadata = load_metadata(model_location)
```

Comment:
Are these two last conditionals the same code?

Reply:
They are called with different ```cv_mode```, so as ```provide_saved_model``` calls ```build_model```, it will behave differently.